internal/weapons/spear/calamity/calamity.go

```diff
@@ -11,66 +11,56 @@ func init() {
 	core.RegisterWeaponFunc("calamityqueller", weapon)
 }
 
+// Gain 12/15/18/21/24% All Elemental DMG Bonus.
+// Obtain Consummation for 20s after using an Elemental Skill, causing ATK to increase by 3.2/4/4.8/5.6/6.4% per second.
+// This ATK increase has a maximum of 6 stacks.
+// When the character equipped with this weapon is not on the field, Consummation's ATK increase is doubled.
 func weapon(char core.Character, c *core.Core, r int, param map[string]int) string {
-	stacks := 0
-	var ctick func(core.Character, *core.Core, int) func()
-	ctick = func(char core.Character, c *core.Core, skillInitF int) func() {
-		return func() {
-			if c.F != skillInitF {
-				return
-			}
-			if c.Status.Duration("consummation-"+char.Name()) <= 0 {
-				stacks = 0
-				return
-			}
-			if stacks < 6 {
-				stacks++
-			}
-			c.Log.Debugw("consummation-"+char.Name()+" buff ticking", "frame", c.F, "event", core.LogCharacterEvent, "stacks", stacks)
-			if stacks == 6 {
-				char.AddTask(ctick(char, c, skillInitF), "consummation-"+char.Name()+"-tick", c.Status.Duration("consummation-"+char.Name()))
-			} else {
-				char.AddTask(ctick(char, c, skillInitF), "consummation-"+char.Name()+"-tick", 60)
-			}
-		}
-	}
 
-	c.Events.Subscribe(core.OnAttackWillLand, func(args ...interface{}) bool {
-		atk := args[1].(*core.AttackEvent)
-		if atk.Info.ActorIndex != char.CharIndex() {
-			return false
-		}
-		if atk.Info.AttackTag != core.AttackTagElementalArt && atk.Info.AttackTag != core.AttackTagElementalArtHold {
-			return false
-		}
-		c.Status.AddStatus("consummation-"+char.Name(), 20*60)
-		ctick(char, c, c.F)()
-		return false
+	m := make([]float64, core.EndStatType)
+	dmg := .09 + float64(r)*.03
+	atkbonus := .024 + float64(r)*.008
+	m[core.PyroP] = dmg
+	m[core.HydroP] = dmg
+	m[core.CryoP] = dmg
+	m[core.ElectroP] = dmg
+	m[core.AnemoP] = dmg
+	m[core.GeoP] = dmg
+	m[core.DendroP] = dmg
 
-	}, fmt.Sprintf("consummation-%v", char.Name()))
+	expiry := -1
+	skillInitF := 0
 
-	val := make([]float64, core.EndStatType)
-	mod := float64(r - 1)
+	c.Events.Subscribe(core.PreSkill, func(args ...interface{}) bool {
+
+		if expiry < c.F {
+			skillInitF = c.F
+		}
+		expiry = c.F + 60*20
+
+		return false
+	}, fmt.Sprintf("calamity-queller-%v", char.Name()))
 
 	char.AddMod(core.CharStatMod{
```

Comment:
it's probably best if this AddMod gets moved into the Subscribe event with a proper expiry set. We're trying to clean up a bit and avoid using -1 (which is just laziness on my part). See https://github.com/genshinsim/gcsim/issues/203

Reply:
i think the AddMod will have to stay because the Ele% is permanent buff.
unless i am missing something here or you meant the ATK buff...
i could try moving the ATK buff inside the event but don't know if i can do that while keeping stacks from previous cast.

Reply:
Oh sorry you're right. Let me think about this for a bit